src/core/qgspointxy.h

```diff
@@ -147,6 +147,9 @@ class CORE_EXPORT QgsPointXY
      */
     QPointF toQPointF() const
     {
+      if ( isEmpty() ) // special case
+        return QPointF();
```

Comment:
Note that a null QPointF is actually QPointF( 0, 0 ). But I can't see how else we could handle this.